lib/galaxy/datatypes/data.py

```diff
@@ -927,6 +927,10 @@ def regex_line_dataprovider(self, dataset, **settings):
         return dataproviders.line.RegexLineDataProvider(dataset_source, **settings)
 
 
+class Directory(Text):
```

Comment:
What's the reason to inherit from Text ?

Reply:
https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/datatypes/text.py#L24 HTML felt like the closest thing and it descends from ``Text`` also. Do you have an alternative suggestion?

Reply:
plain Data, I think composite datatypes wouldn't have to be restricted to text files ? Or maybe they are ?